.github/workflows/pull_request.yml

```diff
@@ -73,8 +73,8 @@ jobs:
           python-version: ${{ env.DEFAULT_PYTHON }}
       - name: Install requirements
         run: |
-          python3 -m pip install -U pip
-          python3 -m pip install -r docker/main/requirements-dev.txt
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          uv pip install --system -r docker/main/requirements-dev.txt
```

Comment:
I'd suggest you install uv with:

```
- name: Install uv
  uses: astral-sh/setup-uv@v3
```

https://docs.astral.sh/uv/guides/integration/github/